### DISCUSSION TITLE
Alert user based on' GeoFences'.

### DISCUSSION THREAD

**Discussion by timbo-barnes** [opengisch/QField#3708] [Q&A]
One thing that would be really helpful for fieldwork with QField is the ability to designate layers as 'geofences':
i.e. a hazard point layer is included in the Qfield project, alert the user when they get within xx meters of a hazard.
or a polygon layer is included in the project representing restricted/no-go areas. Qfield app alerts the user when they enter the area.

**Comment by suricactus** [comment]
Thanks for the suggestion. This feature is unlikely to be developed in QField without financial support for it, if you are interested please drop us an email on info at opengis.ch.

In the meantime, I believe with some ninja skills in QGIS expressions you can make the screen to blink and flash whenever the condition you mentioned is satisfied.

**Comment by ntrip6** [comment]
Note that geofences are available in Mergin Input
https://www.lutraconsulting.co.uk/blog/2021/10/06/input-v1-release/
and are an open enhancement in QGIS
https://github.com/qgis/QGIS-Enhancement-Proposals/issues/217

**Reply by nirvn** [reply]
@ntrip6 , the geofences hack you are referring to with input is something QField has also supported for years. The relevant variables to build the default value expression are listed in QField's documentation site here (https://docs.qfield.org/how-to/gnss/#positioning-variables). That's however a limited workaround and doesn't offer any visual indication _prior to_ getting to a feature form, which is partly why I'm excited about upcoming QField 3.4's geofencing functionality.

**Comment by nirvn** [comment]
Geofencing has been implemented and will be available in the next version of QField (3.4). The PRs covering the implementation is https://github.com/opengisch/QField/pull/5409 . The implementation includes a visual warnings when entering / leaving a geofenced area as well as a digitizing prohibition when in or out of geofenced areas. Pretty exciting functionality! :)

**Reply by timbo-barnes** [reply]
Fantastic, thanks QField team :)
